test/refract/data/test-Extend.cc

```diff
@@ -0,0 +1,209 @@
+                make_element<StringElement>("bar"),
+                make_element<StringElement>("baz")));
+
+        auto third = make_empty<StringElement>();
+        first->attributes().set("samples",
+            make_element<ArrayElement>( //
+                make_element<StringElement>("zur"),
+                make_element<StringElement>("zul")));
+
+        auto extend = make_element<ExtendElement>();
+        extend->get().push_back(std::move(first));
+        extend->get().push_back(std::move(second));
+        extend->get().push_back(std::move(third));
+
+        WHEN("it is merged")
+        {
+            auto merged = extend->get().merge();
+
+            THEN("the result is not null")
+            {
+                REQUIRE(merged);
+
+                THEN("the result is a string element")
+                {
+                    REQUIRE(dynamic_cast<const StringElement*>(merged.get()));
+                }
+
+                THEN("the result is an empty element")
+                {
+                    REQUIRE(merged->empty());
+                }
+
+                THEN("the result element has one attribute")
+                {
+                    REQUIRE(merged->attributes().size() == 1);
+                }
+
+                THEN("the result element contains an attribute with key `samples`")
+                {
+                    auto attr = merged->attributes().find("samples");
+                    REQUIRE(attr != merged->attributes().end());
+
+                    const IElement* samples = attr->second.get();
+
+                    THEN("that attribute is not nullptr")
+                    {
+                        REQUIRE(samples);
+
+                        THEN("that attribute is an array element")
+                        {
+                            auto samplesArray = dynamic_cast<const ArrayElement*>(samples);
+                            REQUIRE(samplesArray);
+
+                            THEN("the array is not empty")
+                            {
+                                REQUIRE(!samplesArray->empty());
+                            }
+
+                            THEN("the array contains two elements")
+                            {
+                                REQUIRE(samplesArray->get().size() == 2);
+                            }
+
+                            THEN("the array contains the string elements `zur` & `zul`]")
+                            {
+                                const IElement* el1 = samplesArray->get().begin()[0].get();
+                                REQUIRE(dynamic_cast<const StringElement*>(el1)->get() == "zur");
+                                const IElement* el2 = samplesArray->get().begin()[1].get();
+                                REQUIRE(dynamic_cast<const StringElement*>(el2)->get() == "zul");
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
```

Comment:
- missing test cases for merging other Elements (just content testing is enough for this cases)
- missing test cases to merge incompatible Elements (e.g. StringElement and NumberElement)hink about impl check
- add incompatible elements while `push_back` 

(yes i know it technical debt)

Reply:
It's technical debt, so let's shoot it down.

Reply:
Ok, let it pass
At least, we should add issue